### PR TITLE
*: support for running parallel suites

### PR DESCRIPTION
**Pull request by zimulala**

### CODE REVIEW

File: run.go [hunk 1]
```diff
@@ -84,8 +86,26 @@ func TestingT(testingT *testing.T) {
...
+	}
+
+	wg := sync.WaitGroup{}
+	rets := make([]*Result, len(allSuites))
+	for i, suite := range allSuites {
+		wg.Add(1)
+		rets[i] = &Result{}
+		go func(ret *Result, s interface{}) {
+			defer wg.Done()
+			ret = Run(s, runConf)
```
Comment (zz-jason) [inline run.go]: assign the returned pointer to the input parameter is not correct, how about this:
```suggestion
			result.Add(Run(s, runConf))
```
and we can remove `rets`

**Comment by zimulala** [comment]
PTAL @zz-jason

File: run.go [hunk 2]
```diff
@@ -84,12 +86,32 @@ func TestingT(testingT *testing.T) {
...
+			result.Add(Run(suite, runConf))
+		}
+		return &result
+	}
+
+	wg := sync.WaitGroup{}
+	suiteRunners := make([]*suiteRunner, 0, len(allSuites))
 	for _, suite := range allSuites {
-		result.Add(Run(suite, runConf))
+		suiteRunners = append(suiteRunners, parallelRun(suite, runConf, &wg))
```
Comment (zz-jason) [inline run.go]: Why add a new function `parallelRun()` instead of reusing the old logic?

Comment (zimulala) [reply run.go]: We need to execute `setUpSuite` serially first. And I hope not to affect the original function.

Comment (zimulala) [reply run.go]: But this can also be used with the original.

File: check.go [hunk 3]
```diff
@@ -619,36 +619,73 @@ func newSuiteRunner(suite interface{}, runConf *RunConf) *suiteRunner {
...
+		runner.tracker.waitAndStop()
+		if runner.keepDir {
+			runner.tracker.result.WorkDir = runner.tempDir.path
+		} else {
+			runner.tempDir.removeAll()
+		}
+		return
+	}
+
+	runner.tracker.start()
```
Comment (crazycs520) [inline check.go]: Should move to line#626 ?

File: run.go [hunk 4]
```diff
@@ -46,6 +47,7 @@ var (
 )
 
 var CustomVerboseFlag bool
+var CustomParallelSuiteFlag bool
```
Comment (winkyao) [inline run.go]: Please add a flag.Bool for it.

**Pull request merged**